src/inc/til/string.h

```diff
@@ -376,6 +396,81 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         return { beg, end };
     }
 
+    // Splits a font-family list into individual font-families. It loosely follows the CSS spec for font-family.
+    // It splits by comma, handles quotes and simple escape characters, and it cleans whitespace.
+    //
+    // This is not the right place to put this, because it's highly specialized towards font-family names.
+    // But this code is needed both, in our renderer and in our settings UI. At the time I couldn't find a better place for it.
+    void iterate_font_families(const std::wstring_view& families, auto&& callback)
+    {
+        std::wstring family;
+        bool backspace = false;
+        bool delayedSpace = false;
+        wchar_t stringType = 0;
+
+        for (const auto ch : families)
+        {
+            if (!backspace)
+            {
+                switch (ch)
+                {
+                case ' ':
+                    if (stringType)
+                    {
+                        // Spaces are treated literally inside strings.
+                        break;
+                    }
+                    delayedSpace = !family.empty();
+                    continue;
+                case '"':
+                case '\'':
+                    if (stringType && stringType != ch)
+                    {
+                        // Single quotes inside double quotes are treated literally and vice versa.
+                        break;
+                    }
+                    stringType = stringType == ch ? 0 : ch;
+                    continue;
+                case ',':
+                    if (stringType)
+                    {
+                        // Commas are treated literally inside strings.
+                        break;
+                    }
+                    if (!family.empty())

```

Comment:
delayedSpace is ignored here - which I believe is correct?

Reply:
Yes, because we want to strip trailing whitespace.